addons/mail/static/tests/composer/composer_tests.js

```diff
@@ -908,3 +908,130 @@ QUnit.test("Message is sent only once when pressing enter twice in a row", async
     // weak test, no guarantee that we waited long enough for the potential second message to be posted
     await contains(".o-mail-Message-content", { text: "Hello World!" });
 });
+
+QUnit.test('display canned response suggestions on typing ":"', async () => {
+    const pyEnv = await startServer();
+    const guestId = pyEnv["mail.guest"].create({ name: "Mario" });
+    const channelId = pyEnv["discuss.channel"].create({
+        name: "test",
+        channel_member_ids: [
+            [0, 0, { partner_id: pyEnv.currentPartnerId }],
+            Command.create({ guest_id: guestId }),
+        ],
+    });
+    pyEnv["mail.shortcode"].create({
+        source: "hello",
+        substitution: "Hello! How are you?",
+    });
+    const { openDiscuss } = await start();
+    await openDiscuss(channelId);
+    await contains(".o-mail-Composer-input");
+    await contains(".o-mail-Composer-suggestionList .o-open", { count: 0 });
+    await insertText(".o-mail-Composer-input", ":");
+    await contains(".o-mail-Composer-suggestionList .o-open");
+    await contains(".o-mail-NavigableList-item", { text: "helloHello! How are you?" });
+});
+
+QUnit.test("use a canned response", async () => {
+    const pyEnv = await startServer();
+    const guestId = pyEnv["mail.guest"].create({ name: "Mario" });
+    const channelId = pyEnv["discuss.channel"].create({
+        name: "test",
+        channel_member_ids: [
+            [0, 0, { partner_id: pyEnv.currentPartnerId }],
+            Command.create({ guest_id: guestId }),
+        ],
+    });
+    pyEnv["mail.shortcode"].create({
+        source: "hello",
+        substitution: "Hello! How are you?",
+    });
+    const { openDiscuss } = await start();
+    await openDiscuss(channelId);
+    await contains(".o-mail-Composer-suggestionList");
+    await contains(".o-mail-Composer-suggestionList .o-open", { count: 0 });
+    await contains(".o-mail-Composer-input", { value: "" });
+    await insertText(".o-mail-Composer-input", ":");
+    await click(".o-mail-Composer-suggestion");
+    await contains(".o-mail-Composer-input", { value: "Hello! How are you? " });
+});
+
+QUnit.test("use a canned response some text", async () => {
+    const pyEnv = await startServer();
+    const guestId = pyEnv["mail.guest"].create({ name: "Mario" });
+    const channelId = pyEnv["discuss.channel"].create({
+        name: "Mario",
```

Comment:
"use a canned response" => "select a canned response suggestion"

"use" is unclear